data_science/terraform/dlami_asg/userdata.sh.tpl

```diff
@@ -18,5 +26,16 @@ c.NotebookApp.password = u'${hashed_password}'
 
 EOF
 
+# Upgrade and install latest versions of modules
+/home/ubuntu/anaconda3/bin/pip install --upgrade pip
+/home/ubuntu/anaconda3/bin/pip install \
+            pillow==5.1.0 \
+            seaborn==0.8.1 \
+            scikit-learn==0.19.1 \
+            tqdm==4.19.7
+
+# Install s3contents
+/home/ubuntu/anaconda3/bin/pip install s3contents==0.2.2
```

Comment:
Better, but we’re only pinning the top-level packages.

e.g. pillow depends on olefile, but we haven't pinned the version of the latter

Yes, packaging sucks. Sorry. 😕